network/p2p/readSubscription.go

```diff
@@ -87,10 +92,56 @@ func (r *readSubscription) receiveLoop(wg *sync.WaitGroup) {
 			return
 		}
 
+		// if pubsub.WithMessageSigning(true) and pubsub.WithStrictSignatureVerification(true),
+		// the emitter is authenticated
+		emitterKey, err := messagePubKey(rawMsg)
+		if err != nil {
+			r.log.Err(err).Msg("failed to extract libp2p public key of message")
+		}
+
+		flowKey, err := FlowPublicKeyFromLibP2P(emitterKey)
+		if err != nil {
+			r.log.Err(err).Msg("failed to extract flow public key of libp2p key")
+			return
+		}
+
```

Comment:
I would also move the [message unmarshalling](https://github.com/onflow/flow-go/pull/1163/files#diff-b25dcc55780477ea9266ec07bf26a1b6c2f10dc6143aed7f60da81b0d861cccbR85-R93) which is a more heavier operation here after the key extraction since why bother to unmarshal if we can't extract the key.

Reply:
Given that we have `pubsub.WithMessageSigning(true)` and `pubsub.WithStrictSignatureVerification(true)`, I would note such a message would not arrive here, but in theory, agreed.